torch/compiler/__init__.py

```diff
@@ -0,0 +1,319 @@
+import torch
+import torch._dynamo
+import torch._inductor
+from typing import Callable, Union, List, Set, Tuple, Any, Dict
+from torch._dynamo.eval_frame import OptimizedModule
+
+__all__ = [
+    "OptimizedModule",
+    "compile",
+    "is_enabled",
+    "reset",
+    "allow_in_graph",
+    "list_backends",
+    "explain",
+    "is_compiling",
+    "disable",
+    "list_options",
+    "list_mode_options",
+]
+
+def compile(*args, **kwargs):
+    """
+    See :func:`torch.compile` for details on the arguments for this function.
+    """
+
+    return torch.compile(*args, **kwargs)
+
+def is_enabled() -> bool:
+    """
+    Check if compilation is enabled in the current environment.
+
+    This function checks if the `torch.compile` attribute exists in the `torch` module,
+    indicating that compilation is enabled. It returns `True` if compilation is enabled,
+    and `False` otherwise.
+
+    Returns:
+    - A boolean value indicating whether compilation is enabled (`True`) or not (`False`).
+
+    Note:
+    - The function checks the presence of the `torch.compile` attribute using the `hasattr()` function.
+    - If `torch.compile` exists, it indicates that compilation is enabled, and the function returns `True`.
+    - If `torch.compile` does not exist, it indicates that compilation is not enabled, and the function returns `False`.
+
+    """
+
+    if hasattr(torch, "compile"):
+        return True
+    else:
+        return False
```

Comment:
When will this function return False?

In current versions of PyTorch this will always be true.  In older versions it won't exist.  I think it is a nice idea, but I'd still suggest users just write `hasattr(torch, "compile")` in their codebase directly.